builtin/providers/clc/resource_clc_group.go

```diff
@@ -0,0 +1,118 @@
+package clc
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/CenturyLinkCloud/clc-sdk"
+	"github.com/CenturyLinkCloud/clc-sdk/group"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCLCGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCLCGroupCreate,
+		Read:   resourceCLCGroupRead,
+		Update: resourceCLCGroupUpdate,
+		Delete: resourceCLCGroupDelete,
+		Schema: map[string]*schema.Schema{
+			"location_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parent": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"parent_group_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"custom_fields": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
+			},
+		},
+	}
+}
+
+func resourceCLCGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clc.Client)
+	dc := d.Get("location_id").(string)
+	m, err := dcGroups(dc, meta)
+	if err != nil {
+		return fmt.Errorf("Failed pulling groups in location %v - %v", dc, err)
+	}
+	name := d.Get("name").(string)
+	// use an existing group if we have one
+	if m[name] != "" {
+		log.Printf("[INFO] Using EXISTING group: %v => %v", name, m[name])
+		d.SetId(m[name])
+		return nil
+	}
```

Comment:
hmm, so I revisited this problem and it's a bit thorny. here are the details:

- there's no way to create a new toplevel group and every group needs a parent
- parent_group_id is required on create
- groups don't have any uniqueness by attribute. eg. you can have multiple groups named foo (under the same parent). so it's not possible to create a resource (with some name) that already exists, a duplicate will always be created. 
- the tests may need a hard-coded guid
- import is not yet available

putting this altogether, it seems to me that having a poor man's import would be ideal. it's entirely possible I'm not understanding the intended pattern in the conventional way. would it be so horrible to break the rules here?